### PR TITLE
feat(template): Adding setuptools as a build-system option

### PR DESCRIPTION
**Pull request by Anavelyz**
## Pull Request description
- Adding documentation about _**setuptools**_.
- Enable option in TUI for _**setuptools**_.
- Adding _**setuptools**_ as a build-system:
  - Creating a `setuptools-pyproject.toml`.
  - Editing `post_gen_project.py`.
  - Editing cookiecutter.json.
  - Editing Makefile of the template.
  - Adding _**setuptools**_ in conda/dev.yaml.
  - Adding _**setuptools**_ to build-system.sh
  - Adding code to base.sh to check if the _**setuptools**_ command is available in the system.
- Standardize the use of literal strings within the template  with double quotes.
- Adding `noqa` in profile.py to ignore ruff warnings

Solve #53

## Pull Request checklists

This PR is a:
- [ ] bug-fix
- [x] new feature
- [x] maintenance

About this PR:
- [x] it includes tests.
- [x] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [x] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:
- [ ] I have reviewed the changes and it contains no misspelling.
- [ ] The code is well commented, especially in the parts that contain more complexity.
- [x] New and old tests passed locally.

## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved.

### CODE REVIEW

File: tests/smoke/base.sh [hunk 1]
```diff
@@ -47,6 +47,8 @@ elif command -v flit &> /dev/null; then
   flit install
 elif command -v meson &> /dev/null; then
   pip install -e .
+elif  command -v setuptools &> /dev/null; then
+  pip install --editable .
```
Comment (Anavelyz) [inline tests/smoke/base.sh:51]: @xmnlab This is the same as for meson, do I leave it as it is now or change it to `pip install -e .`?

Comment (Anavelyz) [inline tests/smoke/base.sh:51]: @xmnlab This is the same as for meson, do I leave it as it is now or change it to `pip install -e .`?

File: tests/smoke/base.sh
Comment (Anavelyz) [inline tests/smoke/base.sh:1]: @xmnlab This is the same as for meson, do I leave it as it is now or change it to `pip install -e .`?

**Comment by Anavelyz** [comment]
@xmnlab In tests/smoke/base.sh there is the same for setuptools as in meson, do I leave it as it is now or change it to `pip install -e .`?

**Comment by xmnlab** [comment]
> @xmnlab In tests/smoke/base.sh there is the same for setuptools as in meson, do I leave it as it is now or change it to `pip install -e .`?

feel free to keep it .. as both are the same .. it is not a problem .. thanks

**Comment by xmnlab** [comment]
@Anavelyz in general looks good to me.
the only problem now is that there is no setuptools command: https://github.com/osl-incubator/scicookie/actions/runs/5456097301/jobs/9928662242?pr=140#step:5:1820
and for that reason it cannot install the command with pip

as setuptool is kind the legacy because in the past we just used to use setuptools and maybe also it is used as dependency for the other build-system, let's use it in the `else` case. and maybe add a comment there ... I will add a comment direct in the file as well

File: tests/smoke/base.sh [hunk 2]
```diff
@@ -47,6 +47,8 @@ elif command -v flit &> /dev/null; then
   flit install
 elif command -v meson &> /dev/null; then
   pip install -e .
+elif  command -v setuptools &> /dev/null; then
```
Comment (xmnlab) [inline tests/smoke/base.sh:50]:
```suggestion
else
    # use setuptools
```

**Comment by xmnlab** [comment]
thanks @Anavelyz

**Comment by Anavelyz** [comment]
@xmnlab In the future, for setuptools, we can add the options to configure it through setup.cfg or setup.py.

**Comment by xmnlab** [comment]
@Anavelyz I hear you, but my impression is that the community is trying to kill setuptools ... so maybe first it would be nice to see if it would be worthy or not
although setuptools is still used massively

**Pull request merged**